dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorRepository.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.edwmigration.dumper.application.dumper.connector.Connector;
+import java.util.ServiceLoader;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class ConnectorRepository {
+
+  private static final ImmutableMap<String, Connector> CONNECTORS;
```

Comment:
If you use the inner-class pattern:

```
private static class Inner { private static final ConnectorRepository INSTANCE = new ConnectorRepository(); }
public static ConnectorRepository getInstance() { return Inner.INSTANCE; }
```

And then make this be an instance field, that will allow a lazier initialization, and (sometimes, limited) GC of the static data. It may also have some race condition advantages, as it's done under the classloader lock. All methods below can then become instance methods on the singleton.